central/sensor/service/pipeline/complianceoperatorresultsv2/pipeline.go

```diff
@@ -0,0 +1,83 @@
+package complianceoperatorresultsv2
+
+import (
+	"context"
+
+	"github.com/pkg/errors"
+	v2 "github.com/stackrox/rox/central/complianceoperator/v2/checkresults/datastore"
+	"github.com/stackrox/rox/central/convert/internaltov2storage"
+	countMetrics "github.com/stackrox/rox/central/metrics"
+	"github.com/stackrox/rox/central/sensor/service/common"
+	"github.com/stackrox/rox/central/sensor/service/pipeline"
+	"github.com/stackrox/rox/central/sensor/service/pipeline/reconciliation"
+	"github.com/stackrox/rox/generated/internalapi/central"
+	"github.com/stackrox/rox/pkg/centralsensor"
+	"github.com/stackrox/rox/pkg/features"
+	"github.com/stackrox/rox/pkg/metrics"
+)
+
+var (
+	_ pipeline.Fragment = (*pipelineImpl)(nil)
+)
+
+// GetPipeline returns an instantiation of this particular pipeline
+func GetPipeline() pipeline.Fragment {
+	return NewPipeline(v2.Singleton())
+}
+
+// NewPipeline returns a new instance of Pipeline.
+func NewPipeline(v2Datastore v2.DataStore) pipeline.Fragment {
+	return &pipelineImpl{
+		v2Datastore: v2Datastore,
+	}
+}
+
+type pipelineImpl struct {
+	v2Datastore v2.DataStore
+}
+
+func (s *pipelineImpl) Capabilities() []centralsensor.CentralCapability {
+	return nil
+}
+
+func (s *pipelineImpl) Reconcile(_ context.Context, _ string, _ *reconciliation.StoreMap) error {
+	// Not currently deleting anything for historical purposes
+	return nil
+}
+
+func (s *pipelineImpl) Match(msg *central.MsgFromSensor) bool {
+	return msg.GetEvent().GetComplianceOperatorResultV2() != nil
+}
+
+// Run runs the pipeline template on the input and returns the output.
+func (s *pipelineImpl) Run(ctx context.Context, clusterID string, msg *central.MsgFromSensor, _ common.MessageInjector) error {
+	defer countMetrics.IncrementResourceProcessedCounter(pipeline.ActionToOperation(msg.GetEvent().GetAction()), metrics.ComplianceOperatorCheckResultV2)
+
+	event := msg.GetEvent()
+	// If a sensor sends in a v1 compliance message we will still process it the v1 way in the event
+	// a sensor is not updated or does not have the flag on.
+	switch event.Resource.(type) {
+	case *central.SensorEvent_ComplianceOperatorResultV2:
```

Comment:
Can you just do a GetComplianceOperatorResultV2() and that does not equal nil instead of the case?

Reply:
Yeah in the other ones I simplified this back to the original pipeline pattern.  I will do that here.